torchtext/csrc/sentencepiece.cpp

```diff
@@ -58,7 +62,8 @@ static auto sentencepiece =
         .def("Encode", &SentencePiece::Encode)
         .def("EncodeAsIds", &SentencePiece::EncodeAsIds)
         .def("EncodeAsPieces", &SentencePiece::EncodeAsPieces)
-        .def("GetPieceSize", &SentencePiece::GetPieceSize)
+        .def("__len__", &SentencePiece::GetPieceSize)
+        .def("__getitem__", &SentencePiece::PieceToId)
```

Comment:
@cpuhrsch Do you think it's make more sense to have `__call__`? This will be callable directly as a transform. Or we could make `EncodeAsIds ` as `__call__`.

Reply:
I think for now this is meant as a simple internal wrapper around this object, which is then wrapped into a functional + transform and as such mirrors the API for the Python sentencepiece model.

Reply:
@mthrok - please correct me if I'm wrong

Reply:
@cpuhrsch Well I just picked up your work, so I do not have a strong preference, but I see `SentencePiece` class as a container/surrogate for the underlying model, and I think natural interpretation of the current implementation is that it is consumed by functionals and transforms.

I do not have strong preference on adding `__call__` but as we see we have `Encode`, `EncodeAsIds` and `EncodeAsPieces` methods for encoding, and users have to pick one appropriate for them, so I do not feel adding `__call__` would provide much value.